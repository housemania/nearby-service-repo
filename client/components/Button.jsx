import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';


const buttonDiv = {
  float: 'left',
  width: '100px',
  height: '500px',
  display: 'flex',
  textAlign: 'center',
  justifyContent: 'center',
  backgroundColor: 'transparent',
  outlineStyle: 'solid'
};
const buttonStyle = {
  backgroundColor: 'Transparent',
  border: 'none',
  overflow: 'hidden',
  outline: 'none',
  float: 'left'
};

const Button = (props) => {
  return (
    <button className='button' style={buttonStyle} onClick={props.buttonClickHandler} value={props.value}> {props.value === 'right' ?<FontAwesomeIcon id='right' icon={faChevronLeft} size='2x'/> : <FontAwesomeIcon icon={faChevronRight} size='2x'/>}</button>
  )
}

export default Button;
