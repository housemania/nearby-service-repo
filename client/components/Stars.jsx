import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar } from '@fortawesome/free-solid-svg-icons';
import { faStar as farStar} from '@fortawesome/free-regular-svg-icons';

const starsStyle = {
  display: 'flex',
  textAlign: 'center',
  justifyContent: 'center',
  display: 'inline',
  whiteSpace: 'nowrap'
};
const starContainer = {
  display: 'inline',
  whiteSpace: 'nowrap'
}

const Stars = (props) => {
  const starArr = [];
  for (var i = 0; i < props.rating; i++) {
    starArr.push(1);
  }
  for (var j = props.rating; j < 5; j++) {
    starArr.push(0);
  }

  return (
    <div className='star' style={starContainer}>
      {starArr.map( (star, index) => {
        return (
        <div key={index} style={starsStyle}>{(star === 1) ? <FontAwesomeIcon icon={faStar} /> : <FontAwesomeIcon icon={farStar} />}</div>
        )
      })}
    </div>
  )
}

export default Stars;