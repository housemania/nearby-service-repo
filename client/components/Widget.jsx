import React from 'react';

class Widget extends React.Component {
  constructor() {
    super()
  }

  render() {
    return (
      <div id='morePlaces'>
        <h2>More places to stay</h2>
      </div>
    )
  }
}

export default Widget;