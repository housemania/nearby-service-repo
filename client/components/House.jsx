import styled from 'styled-components';

import React from 'react';
import Stars from './Stars.jsx';
import Heart from './Heart.jsx';

const ImageHouse = styled.img`
  max-width: 100%;
  max-height: 100%;
`;
const ImageContainer = styled.div`
  width: 300px;
  height: 225px;
  position: relative;
  object-fit: scale-down;
`;
const HouseContainer = styled.div`
  float: left;
  margin: 8px 8px 8px 8px;
  width: 300px;
  height: 275x;
`;
const Words = styled.div`
  font-weight: 300;
  padding: 0px 0px 0px 0px;
  margin: 0px 0px 0px 0px;
`;
const LightDescription = styled.div`
  line-height: 30%;
  color: #616A6B;
  font-size: 13px;
`;

class House extends React.Component {
  handleClick() {
    this.props.changeCurrentHouse(this.props.house.nearbyNum);
  }

  render() {
    return (
      <HouseContainer className='house'>
        <ImageContainer>
          <Heart heartArr={this.props.heartArr} arrIndex={this.props.house.arrIndex} heartHouseClicked={this.props.heartHouseClicked}/>
          <ImageHouse
            className='image'
            onClick={this.handleClick.bind(this)}
            src={this.props.house.imgUrl}>
          </ImageHouse>
        </ImageContainer>
        <Words onClick={this.handleClick.bind(this)}>
          <LightDescription className='layout'>
            <h5>
              {this.props.house.type.toUpperCase()}: {this.props.house.location.toUpperCase()}
            </h5>
          </LightDescription>
          <div className='description'>
            <h3>{this.props.house.title} </h3>
          </div>
          <LightDescription className='cost'>
            <h5>{this.props.house.cost} </h5>
          </LightDescription>
          <div className='stars'>
            <h5>
              <Stars
                key={this.props.house.id}
                rating={this.props.house.stars}
              />
              {this.props.house.reviewCount}
            </h5>
          </div>
        </Words>
      </HouseContainer>
    );
  }
}

export default House;
