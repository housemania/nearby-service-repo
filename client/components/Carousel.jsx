import React from 'react';
import styled from 'styled-components';
import House from './House.jsx';
import Button from './Button.jsx';

const CarouselContainer = styled.div`
  width: 1500px;
  height: 500px;
`

const ButtonContainer = styled.div`
  float: left;
  width: 100px;
  height: 500px;
  display: flex;
  textAlign: center;
  justifyContent: center;
`


class Carousel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      startIndex: 0,
      endIndex: 3,
    }
  }

  getDisplayHouses(start, end) {
    this.setState( {
      displayHouses: this.props.nearbyHouseList.slice(start, end)
    }) 
  }

  componentDidMount() {
    this.getDisplayHouses(this.state.startIndex, this.state.endIndex);
  }

  buttonClickHandler(event) {
    this.shiftDisplay(event.target.value);
  }

  shiftDisplay(side) {
    if (side === 'right') {
      let newStart = this.state.startIndex - 1;
      let newEnd = this.state.endIndex - 1;
      this.setState({
        startIndex: newStart,
        endIndex: newEnd
      })
      this.getDisplayHouses(newStart, newEnd);
    } else if (side === 'left') {
      let newStart = this.state.startIndex + 1;
      let newEnd = this.state.endIndex + 1;
      this.setState({
        startIndex: newStart,
        endIndex: newEnd
      })
      this.getDisplayHouses(newStart, newEnd);
    }
  }

  render() {
    return (
      <CarouselContainer className='carousel'>
        <ButtonContainer className='buttonDiv'>
          {this.state.startIndex === 0 ? <ButtonContainer></ButtonContainer> : <Button value='right' buttonClickHandler={this.buttonClickHandler.bind(this)}/>}
        </ButtonContainer>
        <div className='houseDisplay'>
          {this.state.displayHouses ? this.state.displayHouses.map((house) => {
            return <House changeCurrentHouse={this.props.changeCurrentHouse} house={house} key={house._id}/>
          }): null}
        </div>
        <ButtonContainer className='buttonDiv'>
          {this.state.endIndex === 12 ? <ButtonContainer></ButtonContainer> : <Button value='left' buttonClickHandler={this.buttonClickHandler.bind(this)}/>}
        </ButtonContainer>
      </CarouselContainer>
    )
  }
}

export default Carousel;