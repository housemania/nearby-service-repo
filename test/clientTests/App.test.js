import { mount, shallow, render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Widget from '../../client/components/Widget.jsx';
import House from '../../client/components/House.jsx';
import Button from '../../client/components/Button.jsx';
import Carousel from '../../client/components/Carousel.jsx';
import Stars from '../../client/components/Stars.jsx';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';


// Widget Component 
describe('Test Widget component functionality', () => {
  const options = {
    disableLifecycleMethods: true
  }

  it('should render correctly', () => {
    const component = shallow(<Widget />, options);
    expect(component).toMatchSnapshot();
  })

  it('should call componentDidMount once',  () => {
    const componentFunc = jest.fn();
    const component = shallow(<Widget/>, options);
    component.instance().getNearbyHouses = componentFunc;
    component.update();
    const instance = component.instance();
    instance.componentDidMount();
    
    expect(instance.getNearbyHouses).toHaveBeenCalledTimes(1);
  })

  it('should render Carousel if state.view is true',  () => { // this one 
    const component = shallow(<Widget />, options);
    component.setState({view: true});
    component.update();
    expect(component.find(Carousel)).toHaveLength(1);
  })

  it('should change currentHouse state to the id changeCurrentHouse is called with',  () => {
    const component = shallow(<Widget />, options);
    component.instance().getNearbyHouses = jest.fn();
    component.update();
    const instance = component.instance();
    instance.changeCurrentHouse(4);
    expect(component.state('currentHouse')).toBe(4);
  })

  it('should change update nearbyHouseList state to array of houses',  () => {
    const houses = ['house1', 'house2', 'house3'];
    const component = shallow(<Widget />, options);
    const instance = component.instance();
    instance.updateHouseList(houses);
    expect(component.state('nearbyHouseList')).toBe(houses);
  })
});

// House Component testing 
describe('Test House component functionality', () => {
  const clickFn = jest.fn();
  const props = {
    imgUrl: "https://housemania.s3-us-west-1.amazonaws.com/annie-spratt-BmjlyHwV1S0-unsplash.jpg",
    cost: '$309/night',
    location: 'Saratoga',
    type: 'apartment',
    title: 'quaint house',
    stars: 4,
    reviewCount: 20
  }; 

  it('should render correctly', () => {
    const component = renderer.create(<House house={props}></House>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('house button click call handleClick to run changeCurrentHouse', () => {
    const component = shallow(<House changeCurrentHouse={clickFn} house={props}/>);
    component
      .find('.image')
      .simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});

// Test Carousel Component 
describe('Test Carousel component functionality', () => {
  const options = {
    disableLifecycleMethods: true
  };
  const clickFn = jest.fn();
  
  it('should render correctly', () => {
    const component = shallow(<Carousel />, options);
    expect(component).toMatchSnapshot();
  })

  it('renders <House /> one times', () => {
    const component = shallow(<Carousel />, options);
    component.setState({
      displayHouses: [
        {
          imgUrl: "https://housemania.s3-us-west-1.amazonaws.com/annie-spratt-BmjlyHwV1S0-unsplash.jpg",
          cost: '$309/night',
          location: 'Saratoga',
          type: 'apartment',
          title: 'quaint house',
          stars: 4,
          reviewCount: 20
        }
      ]
    })
    expect(component.find(House)).toHaveLength(1);
  })

  it('renders <Button /> two times for normal display', () => {
    const component = shallow(<Carousel />, options);
    component.setState({
      endIndex: 5,
      startIndex: 2
    })
    expect(component.find('.buttonDiv')).toHaveLength(2);
  })

  it('should increase state.startIndex by 1 when shiftDisplay called with left',  () => {
    const component = shallow(<Carousel />, options);
    component.instance().getDisplayHouses = clickFn;
    component.update();
    const instance = component.instance();
    expect(component.state('startIndex')).toBe(0);
    instance.shiftDisplay('left');
    expect(component.state('startIndex')).toBe(1);
  })

  it('should call shiftDisplay function with a side',  () => {
    const component = shallow(<Carousel />, options);
    component.instance().shiftDisplay = clickFn;
    component.update();
    component.instance().shiftDisplay('left');
    expect(component.instance().shiftDisplay).toHaveBeenCalledWith('left');
  })

  it('should call getDisplayHouses function with a start and end index',  () => {
    const component = shallow(<Carousel />, options);
    component.instance().getDisplayHouses = clickFn;
    component.update();
    component.instance().getDisplayHouses(1, 3);
    expect(component.instance().getDisplayHouses).toHaveBeenCalledWith(1,3);
  })

  it('should call componentDidMount once',  () => {
    const componentFunc = jest.fn();
    const component = shallow(<Carousel/>, options);
    component.instance().getDisplayHouses = componentFunc;
    component.update();
    const instance = component.instance();
    instance.componentDidMount();
    
    expect(instance.getDisplayHouses).toHaveBeenCalledTimes(1);
  })

});


// Test Button Component 
describe('Button component functionality', () => {
  const clickFn = jest.fn();
  const props = {
    buttonClickHandler: jest.fn(),
    value: 'right'
  }

  it('should render correctly', () => {
    const component = renderer.create(<Button props={props}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('button value should be right', () => { 
    const component = shallow(<Button value={'right'}/>);
    component
    expect(component.prop('value')).toEqual('right');
  });

  it('button click call handleClick to run shift the display', () => {
    const component = shallow(<Button buttonClickHandler={clickFn}/>);
    component
      .find('.button')
      .simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});

// Test Stars Component
describe('Stars component functionality', () => {
  it('should render correctly', () => {
    const component = renderer.create(<Stars/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders <FontAwesomIcon /> five times for every rating', () => {
    const component = shallow(<Stars />);
    component.setProps({
      rating: 4
    })
    expect(component.find(FontAwesomeIcon)).toHaveLength(5);
  })
});