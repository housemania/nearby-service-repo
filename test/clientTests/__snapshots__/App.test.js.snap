// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component functionality should render correctly 1`] = `
<button
  className="button sc-ifAKCX keVeUP"
>
   
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-chevron-right fa-w-10 fa-2x "
    data-icon="chevron-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;

exports[`Stars component functionality should render correctly 1`] = `
<div
  className="star sc-bdVaJa gcmXMp"
/>
`;

exports[`Test Carousel component functionality should render correctly 1`] = `
<styled.div
  className="carousel"
>
  <styled.div
    className="buttonDiv"
  >
    <styled.div />
  </styled.div>
  <div
    className="houseDisplay"
  />
  <styled.div
    className="buttonDiv"
  >
    <Button
      buttonClickHandler={[Function]}
      value="left"
    />
  </styled.div>
</styled.div>
`;

exports[`Test House component functionality should render correctly 1`] = `
<div
  className="house sc-bxivhb ghdJAk"
  onClick={[Function]}
>
  <div
    className="image"
  >
    <img
      className="sc-htpNat inkXUF"
      src="https://housemania.s3-us-west-1.amazonaws.com/annie-spratt-BmjlyHwV1S0-unsplash.jpg"
    />
  </div>
  <div
    className="layout"
  >
    <h5>
      apartment
      : 
      Saratoga
       
    </h5>
  </div>
  <div
    className="description"
  >
    <h3>
      quaint house
       
    </h3>
  </div>
  <div
    className="cost"
  >
    <h5>
      $309/night
       
    </h5>
  </div>
  <div
    className="stars"
  >
    <h5>
      <div
        className="star sc-bdVaJa gcmXMp"
      >
        <div
          className="sc-bwzfXH dHMfna"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          className="sc-bwzfXH dHMfna"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          className="sc-bwzfXH dHMfna"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          className="sc-bwzfXH dHMfna"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          className="sc-bwzfXH dHMfna"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="far"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
       
      20
       
    </h5>
  </div>
</div>
`;

exports[`Test Widget component functionality should render correctly 1`] = `
<div
  id="morePlaces"
>
  <h2>
    More places to stay
  </h2>
</div>
`;
